container-storage-setup.sh

```diff
@@ -755,26 +753,33 @@ create_partition() {
     Fatal "udevadm settle after partition creation failed. Exiting."
   fi
 
-  if ! wait_for_dev ${dev}1; then
-    Fatal "Partition device ${dev}1 is not available"
+  part=$(dev_query_first_child $dev)
+
+  if ! wait_for_dev ${part}; then
+    Fatal "Partition device ${part} is not available"
   fi
 }
 
+dev_query_first_child() {
+  lsblk -npl -o NAME "$1" | tail -n +2 | head -1
```

Comment:
I had given a comment on this to simplify it. can you please take care of that as well.